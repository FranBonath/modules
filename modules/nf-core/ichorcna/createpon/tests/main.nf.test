// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test ichorcna/createpon
nextflow_process {

    name "Test Process ICHORCNA_CREATEPON"
    script "../main.nf"
    process "ICHORCNA_CREATEPON"

    tag "modules"
    tag "modules_nfcore"
    tag "ichorcna"
    tag "ichorcna/createpon"

    test("hg19 - one file") {

        when {
            process {
                """
                input[0] = file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/MBC_315.ctDNA.reads.wig", checkIfExists: true)
                input[1] = file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/gc_hg19_1000kb.wig", checkIfExists: true)
                input[2] = file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/map_hg19_1000kb.wig", checkIfExists: true)
                input[3] = file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/GRCh37.p13_centromere_UCSC-gapTable.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.rds.get(0)).name).match("PoN_median.rds") },
                { assert snapshot(file(process.out.txt.get(0)).name).match("PoN_median.txt") },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }


    test("hg19 - two files") {

        when {
            process {
                """
                input[0] = [ file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/MBC_315.ctDNA.reads.wig", checkIfExists: true),
                            file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/MBC_315_T2.ctDNA.reads.wig", checkIfExists: true)
                            ]

                input[1] = file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/gc_hg19_1000kb.wig", checkIfExists: true)
                input[2] = file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/map_hg19_1000kb.wig", checkIfExists: true)
                input[3] = file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/GRCh37.p13_centromere_UCSC-gapTable.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.rds.get(0)).name).match("PoN_median.rds") },
                { assert snapshot(file(process.out.txt.get(0)).name).match("PoN_median.txt") },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }


}
