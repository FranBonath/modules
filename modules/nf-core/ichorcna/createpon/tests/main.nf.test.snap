{
    "hg19 - one file": {
        "content": [
            [
                "versions.yml:md5,59dbc83520b9e00a736f49ed2513657a"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-08-03T07:09:23.96541049"
    },
    "hg19 - two files": {
        "content": [
            [
                "versions.yml:md5,59dbc83520b9e00a736f49ed2513657a"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-08-03T07:09:46.582232806"
    }
}